source/common/tracing/opentracing_driver_impl.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/tracing/http_tracer.h"
+
+#include "common/common/logger.h"
+
+#include "opentracing/tracer.h"
+
+namespace Envoy {
+namespace Tracing {
+
+class OpenTracingSpan : public Span, Logger::Loggable<Logger::Id::tracing> {
+public:
+  OpenTracingSpan(bool use_single_header_propagation, bool use_tracer_propagation,
+                  std::unique_ptr<opentracing::Span>&& span);
+
+  // Tracing::Span
+  void finishSpan() override;
+  void setOperation(const std::string& operation) override;
+  void setTag(const std::string& name, const std::string& value) override;
+  void injectContext(Http::HeaderMap& request_headers) override;
+  SpanPtr spawnChild(const Config& config, const std::string& name, SystemTime start_time) override;
+
+private:
+  bool use_single_header_propagation_;
```

Comment:
I think this and what follows can be const

Reply:
Why can't this be an enum? Do we need 2 booleans?

Reply:
I could change to an enum, but I would think the options could vary independently for a tracer (doesn't zipkin, for example, propagate using both methods?)

Reply:
Perhaps, but would it ever run in both modes? I think it would be driven by config in which one mode is chosen. Enum feels more logical to me but I will admit I don't have all the context you do.

Reply:
Changed to use an enum.